src/rust/grapl-utils/src/iter_ext/mod.rs

```diff
@@ -0,0 +1,60 @@
+pub trait GraplIterExt: Iterator {
+    /// Provides an 'owned' variant to `Itertools::Itertools`'s `::chunks` method.
+    /// Returns an iterator that contains Vec<I::Item> each, at most, the size of `chunk_size`
+    /// While the length of the individual Vec's are guaranteed to be equal or less-than `chunk_size`
+    /// the capacity is not; adding to a Vec may trigger an allocation.
+    fn chunks_owned(self, chunk_size: usize) -> ChunkedIterator<Self>
+    where
+        Self: Sized + Iterator,
+    {
+        ChunkedIterator::new(self, chunk_size)
+    }
+}
+
+pub struct ChunkedIterator<I>
+where
+    I: Iterator + Sized,
+{
+    inner: I,
+    chunk_size: usize,
+}
+
+impl<I> ChunkedIterator<I>
```

Comment:
Is `.chunks` not suitable?

Reply:
Nah, chunks only produces an iterator of iterators where the inner iterators reference a `&T` which causes some lifetime problems when invoking the `upsert_graph` method from the graph-merger. It was much easier to just chunk and own the data.